client/components/data/query-domain-owner-username/index.ts

```diff
@@ -0,0 +1,51 @@
+import { useSelector } from 'react-redux';
+import { useQuerySitePurchases } from 'calypso/components/data/query-site-purchases';
+import useUsersQuery from 'calypso/data/users/use-users-query';
+import { getSitePurchases } from 'calypso/state/purchases/selectors';
+import type { SiteDetails } from '@automattic/data-stores';
+import type { ResponseDomain } from 'calypso/lib/domains/types';
+import type { InfiniteData } from 'react-query';
+
+type User = {
+	login: string;
+	linked_user_ID: number;
+};
+
+type UsersData = {
+	users: User[];
+};
+
+export function useDomainOwnerUserName(
+	selectedSite: SiteDetails | null | undefined,
+	domain: ResponseDomain | null | undefined
+): string {
+	useQuerySitePurchases( selectedSite?.ID ?? -1 );
+
+	const purchases = useSelector( ( state ) => getSitePurchases( state, selectedSite?.ID ) );
+
+	const selectedSubscriptionId = domain?.subscriptionId ?? '0';
+
+	const domainSubscription = purchases.filter(
+		( purchase ) => purchase.id === parseInt( selectedSubscriptionId )
+	)[ 0 ];
```

Comment:
```suggestion
	const domainSubscription = purchases.find(
		( purchase ) => purchase.id === parseInt( selectedSubscriptionId )
	);
```
I think we could use `find` here instead of `filter`